admin_manual/issues/general_troubleshooting.rst

```diff
@@ -250,30 +250,149 @@ Service discovery
+
+These *well known* URLs are located under the web root driectory  of your webserver, whether or not your nextcloud
+installation is installed in the root or a subfolder of your web server:
+
+| ``https://example.com/.well-known/``
+|
+
+To check if these *well known* URLs are working correctly on your server, you can go to the ``Overview`` page of the
+``Administrative Settings`` of your Nextcloud installation, e.g. at ``https://example.com/settings/admin/overview``
+and see if there is any warning  shown under the ``Security & setup warnings`` section. Additionlly, the specific sections
+below show how to verify the correct configuration from a linux command-line (see *Tip* boxes).
+
+.. _service-discovery-caldav-and-carddav-label:
+
+CalDAV and CardDAV
+^^^^^^^^^^^^^^^^^^
+
 If you want to use CalDAV or CardDAV clients or other clients that require service discovery
 together with Nextcloud it is important to have a correct working setup of the following
 URLs:
 
 | ``https://example.com/.well-known/carddav``
 | ``https://example.com/.well-known/caldav``
-|
 
-Those need to be redirecting your clients to the correct endpoints. If Nextcloud
-is running at the document root of your Web server the correct URL is
-``https://example.com/remote.php/dav`` for CardDAV and CalDAV and if running in a
-subfolder like ``nextcloud``, then ``https://example.com/nextcloud/remote.php/dav``.
+Those need to be redirecting your clients to the correct endpoints:
+
+- ``https://example.com/remote.php/dav`` if Nextcloud is running at the document root of your Web server, and
+- ``https://example.com/nextcloud/remote.php/dav`` if running in a subfolder like ``nextcloud``.
+
+.. Tip::
+   To check if your server is set up correctly, use the following command with your correct URL for ``YOUR_SERVER``:
+
+   **CalDAV** ::
+
+      YOUR_SERVER=https://example.com
+      SERVICE=caldav
+      curl -Sv --fail -o /dev/null --GET $YOUR_SERVER/.well-known/$SERVICE 2>&1 \
+       | grep -iE '^(> get|< (HTTP/|location))'
+
+   This should return an output similar to this::
+
+         > GET /.well-known/carddav HTTP/2
+         < HTTP/2 301
+         < location: https://example.com/remote.php/dav/
+
+   **CardDav** ::
+
+         YOUR_SERVER=https://example.com
+         SERVICE=carddav
+         curl -Sv --fail -o /dev/null --GET $YOUR_SERVER/.well-known/$SERVICE 2>&1 \
+          | grep -iE '^(> get|< (HTTP/|location))'
+
+   This should return an output similar to this::
+
+         > GET /.well-known/carddav HTTP/2
+         < HTTP/2 301
+         < location: https://example.com/remote.php/dav/
+
+.. _service-discovery-webfinger-and-nodeinfo-label:
+
+WEBFINGER and NODEINFO (and others)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since version 21 of Nextcloud server, these two endpoints
+
+| ``https://example.com/.well-known/webfinger``
+| ``https://example.com/.well-known/nodeinfo``
+
+need to be redirected to the ``index.php`` like so:
```

Comment:
As explained above, this is misleading because it's different from our default implementation. webfinger and nodeinfo however are fine with a chain of redirects.